### PR TITLE
Fixing entropy calculation

### PR DESCRIPTION
**Pull request by abaheti95**
I made a mistake when logging entropy by only looking at the actions rather than the entire distribution.
This fixes that mistake and now does correct entropy calculation

[mlflow of old and new entropy logging](https://dbc-559ffd80-2bfc.cloud.databricks.com/ml/experiments/1411130483624921?o=7395834863327820&searchFilter=&orderByKey=attributes.start_time&orderByAsc=false&startTime=ALL&lifecycleFilter=Active&modelVersionFilter=All+Runs&datasetsFilter=W10%3D&compareRunsMode=CHART)

## Update
I've modified the PR to include more functionalities:
1. Getting the sequence and token entropy of the sequences
2. seq entropy becomes the loss term
3. Token entropy is used to get different percentiles of token entropies throughout the batch

### CODE REVIEW

**Review by dakinggg — COMMENTED** [review]
Since fixing a bug in a narrow, fairly self contained piece of the codebase, lets add unit tests while we're here?

**Comment by gupta-abhay** [comment]
> Since fixing a bug in a narrow, fairly self contained piece of the codebase, lets add unit tests while we're here?

+1 to this

**Comment by abaheti95** [comment]
Okay... modified the code to calculate both token and sequence entropies and added unit tests for both.
Also added piping for entropy loss from @jdchang1 's [entropy loss branch](https://github.com/databricks/compose-rl/tree/jchang/entropy-grpo)

**Review by jdchang1 — APPROVED** [review]
LGTM! Thanks for the fix and tests

**Review by gupta-abhay — COMMENTED** [review]
mostly lg, some nits!

File: compose_rl/utils/utils.py [hunk 1]
```diff
@@ -135,23 +135,70 @@ def get_log_probs(
...
+    Args:
+        logits (torch.Tensor): Logits tensor of shape (batch_size, seq_len, vocab_size).
+        action_mask (torch.Tensor): Mask tensor of shape (batch_size, seq_len) where 1 indicates
+                                    tokens to include in the entropy calculation.
+
+    Returns:
+        torch.Tensor: Entropy values for each item in the batch (batch_size,).
+    """
+    # Calculate per-token entropies
+    token_entropies = get_token_entropies(logits)
```
[the remark this answers is not in the document]
Comment (abaheti95) [reply compose_rl/utils/utils.py]: Thanks for the suggestion... I updated the `get_sequence_entropies` to now ingest `token_entropies` directly
